File: JS/Sorting/quickSort.js
ui;let arr=[9,8,7,6,5,4,3,2,1];

function partition(arr,start,end){
let pivot= arr[start]
let i= start;
let j = end;
while(i<j){
while(i<end && arr[i]<=pivot){
    i++;
}
while(j>start && arr[j]>pivot) j--;

if(i<j){
    [arr[i],arr[j]]=[arr[j],arr[i]];
}


}
[arr[start],arr[j]]=[arr[j],arr[start]]
return j;

}
function quickSort(arr,start,end){
if(start>=end){
    return
}
let p = (partition(arr,start,end))
quickSort(arr,start,p-1)
quickSort(arr,p+1,end)

}
quickSort(arr,0,arr.length-1)

for(a of arr){
    process.stdout.write(`${a} `)
}

File: C++/Sorting/quickSort.cpp
#include<bits/stdc++.h>
using namespace std;
int partition(int arr[],int start,int end){
    int pivot = arr[start];
    int i=start;
    int j=end;
    while(i <j){
        while(arr[i]<=pivot && i<end) i++;
        while(arr[j]>pivot && j>start) j--;
        if(i<j){
            swap(arr[i],arr[j]);
        }
    
    }
swap(arr[start],arr[j]);
return j;

}
void quickSort(int arr[],int start,int end){
if(start>=end) return;
int p = partition(arr,start,end);
 quickSort(arr,start,p-1);
 quickSort(arr,p+1,end);

}
int main(){
    int arr[]={9,8,7,6,5,4,3,2,1};
    int sizeOfArr=sizeof(arr)/sizeof(arr[0]);
    quickSort(arr,0,sizeOfArr-1);
    for(auto it:arr){
        cout<<it<<" ";
    }
}